src/intersection/first-solution/intersection.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Creates an array of values that are in both the first and the second arrays.
+ *
+ * @param {Array} [firstArr=[]] - First array to be compared for repeated values.
+ * @param {Array} [secondArr=[]] - Second array to be compared for repeated values.
+ * @returns {Array} Returns the new array with only intersected values.
+ */
+
+export const arrayIntersection = (firstArr = [], secondArr = []) => {
+  let newArrayIntersected = [];
+  // Test if the arguments are arrays and make sure that arguments are passed
+  if (!Array.isArray(firstArr) || !Array.isArray(secondArr)) {
+    throw new TypeError('The arguments must be arrays');
+  } else if (!firstArr.length || !secondArr.length) {
+    throw new Error('Must pas two arrays to intercept the values');
+  } else {
+    // test if arrays have the same values, return only the values matched
+    newArrayIntersected = [...new Set(firstArr.filter((x) => secondArr.includes(x)))];
+  }
+  return newArrayIntersected;
```

Comment:
will this work? 
delete line 10;
delete line 16 else part:
delete line 18:
line 20: return firstArr.filter((x) => secondArr.includes(x);

Reply:
There are many several ways to refactor a code, all are valid, even in yours I can make it in one line of code and more readable, but that doesn't mean that yours isn't valid, accept suggestions can be valid according to your criteria, you can see some of your suggestions adapted in my code, you don't have to expect that I write the code exactly as you want, your role is to check that everything is ok, and of course make suggestions to improve the code.